src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -318,6 +324,24 @@ public Task<PingReply> SendPingAsync(IPAddress address, int timeout, byte[] buff
             return SendPingAsyncInternal(address, timeout, buffer, options);
         }
 
+        public Task<PingReply> SendPingAsync(IPAddress address, TimeSpan timeout, byte[]? buffer = null,
+            PingOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            Task<PingReply> task = SendPingAsync(address, ToTimeoutMilliseconds(timeout), buffer ?? DefaultSendBuffer, options);
+
+            return task.WaitAsync(cancellationToken);
+        }
+
+        public Task<PingReply> SendPingAsync(string hostNameOrAddress, TimeSpan timeout, byte[]? buffer = null,
+            PingOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            Task<PingReply> task = SendPingAsync(hostNameOrAddress, ToTimeoutMilliseconds(timeout), buffer ?? DefaultSendBuffer, options);
+
+            return task.WaitAsync(cancellationToken);
+        }
```

Comment:
Actually, these implementations are wrong.  They should be removed until the right behaviors can be implemented.  It's not ok to cancel the wait but not cancel the actual operation that's in flight.

Reply:
Indeed
https://github.com/dotnet/runtime/blob/4019e83878a81465f6e42e8502b53bc5d1752f81/src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs#L366

@deeprobin please remove the new Ping APIs that take cancelation tokens and open a new issue to implement them (we can mark it api approved of course). Someone will have to wire this through for all OS's which will be a bit of work, then add this code back.

Reply:
See #67253 - Locally, everything has worked for now - Now it would be nice if the CI passes.

Reply:
New Issue: https://github.com/dotnet/runtime/issues/67260